crates/proc-macro-srv/src/server/rust_analyzer_span.rs

```diff
@@ -70,11 +71,22 @@ impl server::FreeFunctions for RaSpanServer {
         &mut self,
         s: &str,
     ) -> Result<bridge::Literal<Self::Span, Self::Symbol>, ()> {
-        // FIXME: keep track of LitKind and Suffix
+        let literal = str_to_lit_node(s).ok_or(())?;
+
+        let kind = literal_to_external(literal.kind()).ok_or(())?;
+
+        // FIXME: handle more than just int and float suffixes
+        let suffix = match literal.kind() {
+            ast::LiteralKind::FloatNumber(num) => num.suffix().map(ToString::to_string),
+            ast::LiteralKind::IntNumber(num) => num.suffix().map(ToString::to_string),
+            _ => None,
+        }
+        .map(|suffix| Symbol::intern(self.interner, &suffix));
+
         Ok(bridge::Literal {
-            kind: bridge::LitKind::Err,
+            kind,
             symbol: Symbol::intern(self.interner, s),
-            suffix: None,
+            suffix,
```

Comment:
The test fails now because we are passing the suffix and the original string to the interner here, the original still has the suffix in it though so we duplicate it. We'll have to adjust `s` when processing the suffix.